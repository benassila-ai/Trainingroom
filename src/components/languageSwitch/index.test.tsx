import { render } from "@testing-library/react";
import LanguageSwitch from "./index";

// Mock the useTranslation hook
jest.mock("react-i18next", () => ({
    useTranslation: () => ({
      t: (key: string) => key,
      i18n: {
        changeLanguage: jest.fn(),
        language: "en",
      },
    }),
  }));

describe("snapshots", () => {
  it("should match snapshot", () => {
    const { container } = render(<LanguageSwitch />);

    expect(container).toMatchSnapshot();
  });
});
