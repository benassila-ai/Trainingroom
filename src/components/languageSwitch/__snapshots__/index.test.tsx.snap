// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots should match snapshot 1`] = `
<div>
  <div
    class="MuiFormControl-root css-1iw3t7y-MuiFormControl-root"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1093hgx-MuiFormLabel-root-MuiInputLabel-root"
      data-shrink="true"
      for="languages"
    >
      languages
    </label>
    <div
      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-sc8y68-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
    >
      <div
        aria-expanded="false"
        aria-haspopup="listbox"
        aria-labelledby="languages"
        class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-w76bbz-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
        id="languages"
        role="combobox"
        tabindex="0"
      >
        English
      </div>
      <input
        aria-hidden="true"
        aria-invalid="false"
        class="MuiSelect-nativeInput css-j0riat-MuiSelect-nativeInput"
        tabindex="-1"
        value="en"
      />
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-lohd6h-MuiSvgIcon-root-MuiSelect-icon"
        data-testid="ArrowDropDownIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
      <fieldset
        aria-hidden="true"
        class="MuiOutlinedInput-notchedOutline css-1ll44ll-MuiOutlinedInput-notchedOutline"
      >
        <legend
          class="css-w1u3ce"
        >
          <span>
            languages
          </span>
        </legend>
      </fieldset>
    </div>
  </div>
</div>
`;
